packages/labs/observers/src/performance_controller.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2017 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+import {ReactiveControllerHost} from '@lit/reactive-element/reactive-controller.js';
+
+export type PerformanceValueCallback = (
+  entries: PerformanceEntryList,
+  observer: PerformanceObserver,
+  entryList?: PerformanceObserverEntryList
+) => unknown;
+
+export interface PerformanceControllerConfig {
+  config: PerformanceObserverInit;
+  callback?: PerformanceValueCallback;
+  skipInitial?: boolean;
+}
+
+export class PerformanceController {
+  private _host: ReactiveControllerHost;
+  private _config: PerformanceObserverInit;
+  private _observer: PerformanceObserver;
+  private _skipInitial = false;
+  private _unobservedUpdate = false;
+  value?: unknown;
```

Comment:
This could be typed by the return type of the callback

Reply:
What would this look like?

Reply:
Something like:

```ts
export interface PerformanceControllerConfig<T> {
  callback?: PerformanceValueCallback<T>;
}
class PerformanceController<T = boolean> {
  value?: T;
  callback: PerformanceValueCallback<T>;
}
```

We can refine the types later too

Reply:
Ran into issues with threading this through in the tests so filed https://github.com/lit/lit/issues/2350 and hoping we can not block on landing this.